client/extensions/woocommerce/store-sidebar/store-ground-control.js

```diff
@@ -27,7 +29,7 @@ const StoreGroundControl = ( { site, translate } ) => {
 			<div className="store-sidebar__ground-control-site">
 				<Site
 					compact
-					site={ site }
+					site={ isPlaceholder ? null : site }
```

Comment:
Agreed. Fixed in https://github.com/Automattic/wp-calypso/pull/14022/commits/d31a67e864813f2aec6b237bd9de6f28c27c9a3f